client-tools/Shell/ShellFeature.cpp

```diff
@@ -190,4 +202,44 @@ void ShellFeature::start() {
   }
 }
 
+void ShellFeature::beginShutdown() {
+  if (_telemetricsHandler != nullptr) {
+    _telemetricsHandler->beginShutdown();
+  }
+}
+
+void ShellFeature::sendTelemetricsToEndpointTestRedirect(
+    VPackBuilder& builder) {
+  if (_telemetricsHandler != nullptr) {
+    _telemetricsHandler->sendTelemetricsToEndpointTestRedirect(builder);
+  }
+}
+
+void ShellFeature::getTelemetricsInfo(VPackBuilder& builder) {
+  if (_telemetricsHandler != nullptr) {
+    _telemetricsHandler->getTelemetricsInfo(builder);
+  }
+}
+
+void ShellFeature::startTelemetrics() {
+  if (_telemetricsHandler != nullptr) {
+    _telemetricsHandler.reset(nullptr);
+  }
```

Comment:
An extra reset is not necessary because `_telemetricsHandler` is assigned to as the next operation.
Assigning a new object to an existing unique_ptr will always discard the old object before.